provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/web/project/ProjectFilterCustomFunctionAction.java

```diff
@@ -0,0 +1,112 @@
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+@Component
+public class ProjectFilterCustomFunctionAction extends CustomFunctionAction<BlackDuckProjectOptions> {
+    private static final String MISSING_PROVIDER_ERROR = "Provider name is required to retrieve projects.";
+    private static final ActionResponse<BlackDuckProjectOptions> NO_PROJECT_OPTIONS = new ActionResponse<>(HttpStatus.OK, new BlackDuckProjectOptions(List.of()));
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ConfigurationAccessor configurationAccessor;
+    private final BlackDuckProvider blackDuckProvider;
+
+    @Autowired
+    public ProjectFilterCustomFunctionAction(AuthorizationManager authorizationManager, DescriptorMap descriptorMap, FieldValidationUtility fieldValidationUtility,
+        ConfigurationAccessor configurationAccessor, BlackDuckProvider blackDuckProvider) {
+        super(ProviderDistributionUIConfig.KEY_CONFIGURED_PROJECT, authorizationManager, descriptorMap, fieldValidationUtility);
+        this.configurationAccessor = configurationAccessor;
+        this.blackDuckProvider = blackDuckProvider;
+    }
+
+    @Override
+    public ActionResponse<BlackDuckProjectOptions> createActionResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
+        String providerName = fieldModel.getFieldValue(ChannelDistributionUIConfig.KEY_PROVIDER_NAME).orElse("");
+        if (StringUtils.isBlank(providerName)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, MISSING_PROVIDER_ERROR);
+        }
+
+        return fieldModel.getFieldValue(ProviderDescriptor.KEY_PROVIDER_CONFIG_ID)
+                   .map(Long::parseLong)
+                   .map(this::getBlackDuckProjectsActionResponse)
+                   .orElse(NO_PROJECT_OPTIONS);
+    }
+
+    private ActionResponse<BlackDuckProjectOptions> getBlackDuckProjectsActionResponse(Long blackDuckGlobalConfigId) {
+        try {
+            Optional<ConfigurationModel> optionalBlackDuckGlobalConfig = configurationAccessor.getConfigurationById(blackDuckGlobalConfigId);
+            if (optionalBlackDuckGlobalConfig.isPresent()) {
+                StatefulProvider statefulProvider = blackDuckProvider.createStatefulProvider(optionalBlackDuckGlobalConfig.get());
+                BlackDuckProperties properties = (BlackDuckProperties) statefulProvider.getProperties();
+
+                Slf4jIntLogger intLogger = new Slf4jIntLogger(logger);
+                BlackDuckHttpClient blackDuckHttpClient = properties.createBlackDuckHttpClient(intLogger);
+                BlackDuckServicesFactory blackDuckServicesFactory = properties.createBlackDuckServicesFactory(blackDuckHttpClient, intLogger);
+
+                ProjectService projectService = blackDuckServicesFactory.createProjectService();
+                // FIXME improve performance
+                List<BlackDuckProjectSelectOption> options = projectService.getAllProjects()
```

Comment:
If we update the blackduck-common version I believe we can now pass in a predicate to filter the project name